printing_auto_base/models/printing_auto.py

```diff
@@ -0,0 +1,132 @@
+                raise UserError(_("Report was not set"))
+
+    def do_print(self, record):
+        self.ensure_one()
+        record.ensure_one()
+
+        behaviour = self.get_behaviour()
+        printer = behaviour["printer"]
+        if not printer:
+            raise UserError(
+                _("No printer configured to print this {}.").format(self.name)
+            )
+
+        if self.nbr_of_copies <= 0:
+            return (printer, 0)
+        if not self._check_condition(record):
+            return (printer, 0)
+
+        count = 0
+        record = self._get_record(record)
+        for content in self._get_content(record):
+            for _n in range(self.nbr_of_copies):
+                printer.print_document(report=None, content=content, **behaviour)
+                count += 1
+        return (printer, count)
+
+    def get_behaviour(self):
+        if self.printer_id:
+            tray = self.printer_tray_id and self.printer_tray_id.system_name
+            return {"printer": self.printer_id, "tray": tray}
+        if self.file_type == "report":
+            return self.report_id.behaviour()
+        if self.label:
+            return {"printer": self.env.user.default_label_printer_id}
+        return self.env["ir.actions.report"]._get_user_default_print_behaviour()
+
+    def _get_record(self, record):
+        if self.field_object:
+            try:
+                return safe_eval(f"obj.{self.field_object}", {"obj": record})
+            except Exception as e:
+                raise ValidationError(
+                    _("The Object field could not be applied because: %s") % str(e)
+                ) from e
+        return record
+
+    def _check_condition(self, record):
+        domain = safe_eval(self.condition, {"env": self.env})
+        return record.filtered_domain(domain)
+
+    def _get_content(self, record):
+        if self.file_type == "report":
+            return [self.generate_data_from_report(record)]
+        if self.file_type == "attachment":
+            return self.generate_data_from_attachments(record)
+        return []
+
+    def _prepare_attachment_domain(self, record):
+        domain = safe_eval(self.attachment_domain)
+        record_domain = [
+            ("res_id", "=", record.id),
+            ("res_model", "=", record._name),
+        ]
+        return expression.AND([domain, record_domain])
+
+    def generate_data_from_attachments(self, record):
+        domain = self._prepare_attachment_domain(record)
+        attachments = self.env["ir.attachment"].search(domain)
+        if not attachments:
+            raise ValidationError(_("No attachment was found."))
+        return [base64.b64decode(a.datas) for a in attachments]
+
+    def generate_data_from_report(self, record):
+        self.ensure_one()
+        data, _ = self.report_id.with_context(
+            must_skip_send_to_printer=True
+        )._render_qweb_pdf(record.id)
```

Comment:
To handle also `qweb-text` reports (useful to print ZPL labels for instance), better to use the generic rendering method:
```suggestion
        )._render(record.id)
```